src/nhs_grid.jl

```diff
@@ -219,6 +253,50 @@ end
     end
 end
 
+# Fully parallel update with atomic push
+function update_grid!(neighborhood_search::GridNeighborhoodSearch{<:Any, :parallel},
+                      coords_fun::Function)
+    (; cell_list) = neighborhood_search
+
+    # Note that we need two separate loops for adding and removing points.
+    # `push_cell_atomic!` only guarantees thread-safety when different threads push at the
```

Comment:
```suggestion
    # `push_cell_atomic!` guarantees thread-safety when different threads push
```

Reply:
This is correct. It only guarantees thread-safety for parallel push, but not when deletes happen as well.

Reply:
but "at the" makes no sense to me, semantically.

Reply:
Ah, oops. Looks like I couldn't decide between "at the same time" and "simultaneously".